lib/db/mysql/index.js

```diff
@@ -27,6 +33,30 @@ function MysqlStore(options) {
     return next();
   };
   this._pool = mysql.createPool(options);
+
+  // Enforce sane defaults on every new connection.
+  // These *should* be set by the database by default, but it's nice
+  // to have an additional layer of protection here.
+  this._pool.on('connection', function(connection) {
```

Comment:
Oh interesting. It seems we could make use of `_getConnection` inside `_query`, and then yes, we can be sure the session is always set correctly.

Currently, it looks like `_getConnection` was written so that we can release the connection after using it, but that in most cases we just use `pool.query` which keeps the connection internal and thus we never worry about releasing it.